pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -89,28 +87,6 @@ func Register(ctx context.Context, management *config.ManagementContext) {
 	p.Clusters.AddLifecycle(ctx, "cluster-provisioner-controller", p)
 	management.Management.Nodes("").AddHandler(ctx, "cluster-provisioner-controller", p.machineChanged)
 
-	local := &rkedialerfactory.RKEDialerFactory{
-		Factory: management.Dialer,
-		Ctx:     ctx,
-	}
-	docker := &rkedialerfactory.RKEDialerFactory{
-		Factory: management.Dialer,
-		Docker:  true,
-		Ctx:     ctx,
-	}
-
-	driver := service.Drivers[service.RancherKubernetesEngineDriverName]
-	rkeDriver := driver.(*rke.Driver)
-	rkeDriver.DockerDialer = docker.Build
-	rkeDriver.LocalDialer = local.Build
-	rkeDriver.WrapTransportFactory = docker.WrapTransport
-	mgmt := management.Management
-	rkeDriver.DataStore = NewDataStore(mgmt.RkeAddons("").Controller().Lister(),
```

Comment:
Remove these fields from the Provisioner struct as well, like `RKESystemImagesLister`. Also there is more logic here related to RKE1 like `isRke1CustomCluster` and parts of `setVersion`, will that be handled in separate PR?

Reply:
updated in PR: https://github.com/rancher/rancher/pull/50599